### ISSUE TITLE
CSS Module files are loaded before global styles in Storybook

### ISSUE THREAD

**Issue by raph90** [mantinedev/mantine#4902]
### What package has an issue

@mantine/core

### Describe the bug

I'm following the instructions for using Mantine 7 with Storybook and Redwood JS. However, when importing global styles like this in my `storybook.preview.js`:

```tsx
import '@mantine/core/styles.css'
import '@mantine/dates/styles.css'
import 'mapbox-gl/dist/mapbox-gl.css'
import 'src/theme/globalStyles/global-styles.css'

import { MantineProvider } from '@mantine/core'

import { RedwoodProvider } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/dist/apollo'

import theme, { themeResolver } from '../src/theme/theme'

// import UserLocationProvider from '../src/contexts/UserLocationContext'

const preview = {
  decorators: [
    (Story) => (
      <RedwoodProvider titleTemplate="%PageTitle | %AppTitle">
        <RedwoodApolloProvider>
          <MantineProvider theme={theme} cssVariablesResolver={themeResolver}>
            {/* <UserLocationProvider> */}
            <Story />
            {/* </UserLocationProvider> */}
          </MantineProvider>
        </RedwoodApolloProvider>
      </RedwoodProvider>
    ),
  ],
}
```
my CSS Modules stylesheets are being loaded before my global stylesheets. 

Please note that I also had to do the following in my postcss.config.js to get mixins to work correctly in my project:

```js
module.exports = {
  plugins: {
    'postcss-import': {},
    'postcss-mixins': {
      mixinsFiles: ['./src/theme/globalStyles/mixins.css'],
    },
    'postcss-preset-mantine': {},
    'postcss-simple-vars': {
      variables: {
        // 400
        'mantine-breakpoint-xs': '25em',
        // 576
        'mantine-breakpoint-sm': '36em',
        // 768
        'mantine-breakpoint-md': '48em',
        // 1024 - for tablets in portrait
        'mantine-breakpoint-lg': '64em',
        // 1200px - for large screens and above
        'mantine-breakpoint-xl': '75em',
      },
    },
  },
}
```

Finally, my `storybook.config.js` (i.e my webpack config) looks like this:

```js
const path = require('path')

async function webpackFinal(config) {
  config.resolve.alias = {
    ...config.resolve.alias,
    'original-dayjs': path.resolve(__dirname, '../../node_modules/dayjs'),
    '@redwoodjs/testing/web-original': path.resolve(
      __dirname,
      '../../node_modules/@redwoodjs/testing/web'
    ),
    'config/decorators': path.resolve(__dirname, 'decorators'),
  }

  config.module.rules = [
    ...config.module.rules,
    {
      test: /\\.css$/,
      use: [
        'style-loader',
        {
          loader: 'css-loader',
          options: {
            importLoaders: 1,
            modules: true,
          },
        },
        'postcss-loader',
      ],
    },
  ]

  return config
}

module.exports = {
  addons: [
    '@storybook/addon-essentials',
    '@storybook/addon-styling',
    'storybook-dark-mode',
  ],
  webpackFinal,
}

```
It's a bit of a mess but I've tried for a while now to get this to work and I can't. I can confirm that this is only a problem with Storybook and webpack - when Vite loads my files on the Redwood dev server things are in the correct order.

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

@mantine/core 7.0.2

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

This could be to do with a number of problems:
- @storybook/addon-styling is deprecated, and I can't see it doing anything in storybook currently. 
- I don't know if I need to specify style loader in my webpack config
- it could be a postcss config issue. Again, I'm not sure :(

**Comment by kevinsmouts** [comment]
Seems to be a similar root cause than this: https://github.com/mantinedev/mantine/issues/4886

**Comment by raph90** [comment]
@kevinsmouts are you using Storybook? I'm not sure where the issue is, but I think it might be to do with the storybook preview.js. If I move the global imports to somewhere else then they get imported correctly.

**Comment by kevinsmouts** [comment]
I'm not using storybook but it looks like the import order is messed up as soon as we're not using the base modules.css implementation. I think going after one issue or the other will untangle the problem for both :)

**Comment by raph90** [comment]
It's very frustrating. I've spent all evening trying to find a solution and can't :(

**Comment by raph90** [comment]
@rtivital did you find the issue? i'd love to know what's causing the ordering of the imports

**Comment by rtivital** [comment]
This issue is not in the scope of the project, styles import order is fully controlled by third-party tools. From Mantine side I can only provide css files with [layers](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer). This feature will be available in 7.1.0 release and will allow to import styles with `@layer mantine`.

**Comment by raph90** [comment]
Thanks @rtivital, I thought as much. I'm going to put an issue on the Redwood page, since I think this might be an issue there.

**Comment by rtivital** [comment]
The issue is resolved in 7.1.0, you can now use CSS layers to control the order of styles – https://mantine.dev/styles/mantine-styles/#css-layers

We cannot do anything else on our side because Mantine provides static css and cannot control how this file is injected by third-party tools. If you think that the issue is not resolved for you, please report it in the issues of the tools you are using.
